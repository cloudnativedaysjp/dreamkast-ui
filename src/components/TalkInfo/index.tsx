export { TalkInfo } from './TalkInfo'
export { TalkInfoLT } from './TalkInfoLT'
