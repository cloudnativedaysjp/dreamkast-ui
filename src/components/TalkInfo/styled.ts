import styled from 'styled-components'

export const Container = styled.div`
  height: 490px;
  background: rgba(255, 255, 255, 0.7);
  padding: 10px;
  border-radius: 10px;
  overflow: scroll;
  &::-webkit-scrollbar {
    display: none;
  }
`

export const OuterContainer = styled.div`
  padding: 0 0 43px 0;
`

export const Title = styled.h2`
  color: #037f8c;
`

export const Content = styled.div`
  padding: 10px 0;
`
