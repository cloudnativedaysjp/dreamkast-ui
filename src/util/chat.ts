import {
  ChatMessage as ChatMessageInterface,
  ChatMessageMessageTypeEnum,
} from '../client-axios'

export class ChatMessageMap extends Map<number, ChatMessageClass> {
  addMessage = (msg: ChatMessageClass): void => {
    if (!msg.id) return

    if (msg.replyTo) {
      const parent = this.get(msg.replyTo)
      if (parent) {
        if (!parent.children) {
          parent.children = new Map<number, ChatMessageClass>()
        }
        parent.children.set(msg.id, msg)
      } else {
        this.set(msg.id, msg)
      }
    } else {
      this.set(msg.id, msg)
    }
  }
}

export class ChatMessageClass implements ChatMessageInterface {
  id?: number
  profileId?: number
  speakerId?: number
  eventAbbr: string
  roomId?: number
  roomType?: string
  createdAt?: string
  body: string
  messageType: ChatMessageMessageTypeEnum
  replyTo?: number
  children?: Map<number, ChatMessageClass>

  constructor(
    id: number,
    profileId: number,
    speakerId: number,
    eventAbbr: string,
    roomId: number,
    roomType: string,
    createdAt: string,
    body: string,
    messageType: ChatMessageMessageTypeEnum,
    replyTo: number,
  ) {
    this.id = id
    this.profileId = profileId
    this.speakerId = speakerId
    this.eventAbbr = eventAbbr
    this.roomId = roomId
    this.roomType = roomType
    this.createdAt = createdAt
    this.body = body
    this.messageType = messageType
    this.replyTo = replyTo
    this.children = new Map<number, ChatMessageClass>()
  }
}
