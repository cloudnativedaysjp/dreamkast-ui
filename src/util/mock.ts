import { Event, Talk, Track } from '../client-axios'

export const MockEvent: Event = {
  id: 1,
  name: 'Observability Conference 2022',
  abbr: 'o11y2022',
  status: 'open',
  theme: 'this is theme',
  about: 'this is about',
  privacy_policy: 'this is privacy policy',
  privacy_policy_for_speaker: 'this is privacy policy for speaker',
  copyright: 'this is copyright',
  coc: 'this is coc',
}

export const Talks: Talk[] = [
  {
    id: 1,
    trackId: 1,
    videoId: '453942665',
    title: 'ほげふが',
    abstract: 'ぴよぴよ',
    speakers: [
      { id: 1, name: 'alice' },
      { id: 2, name: 'bob' },
    ],
    dayId: 1,
    showOnTimetable: true,
    startTime: '2021-02-15 14:00:00',
    endTime: '2021-02-15 15:00:00',
    talkDuration: 1,
    talkDifficulty: 'hoge',
    talkCategory: 'fuga',
  },
  {
    id: 2,
    trackId: 1,
    videoId: '450845161',
    title: 'ふー',
    abstract: 'ばー',
    speakers: [
      { id: 3, name: 'taro' },
      { id: 4, name: 'jiro' },
    ],
    dayId: 1,
    showOnTimetable: true,
    startTime: '2021-02-15 14:00:00',
    endTime: '2021-02-15 15:00:00',
    talkDuration: 1,
    talkDifficulty: 'hoge',
    talkCategory: 'fuga',
    onAir: true,
  },
  {
    id: 3,
    trackId: 1,
    videoId: '453963160',
    title: 'ほげほげほげ',
    abstract: 'ぴよぴよぴよ',
    speakers: [
      { id: 5, name: 'foo' },
      { id: 6, name: 'bar' },
    ],
    dayId: 1,
    showOnTimetable: true,
    startTime: '2021-02-15 14:00:00',
    endTime: '2021-02-15 15:00:00',
    talkDuration: 1,
    talkDifficulty: 'hoge',
    talkCategory: 'fuga',
  },
]

export const Tracks: Track[] = [
  {
    id: 10,
    name: 'A',
    videoId: '453942665',
    videoPlatform: 'vimeo',
  },
  {
    id: 11,
    name: 'B',
    videoId: '450845161',
    videoPlatform: 'vimeo',
  },
  {
    id: 12,
    name: 'C',
    videoId: '453963160',
    videoPlatform: 'vimeo',
  },
]
