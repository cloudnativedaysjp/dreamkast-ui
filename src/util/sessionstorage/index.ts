// https://developer.mozilla.org/ja/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API

export function isStorageAvailable(storageType: string): boolean | undefined {
  let storage
  try {
    storage = window[storageType]
    const x = '__storage_test__'
    storage.setItem(x, x)
    storage.removeItem(x)
    return true
  } catch (e) {
    return (
      e instanceof DOMException &&
      // everything except Firefox
      (e.code === 22 ||
        // Firefox
        e.code === 1014 ||
        // test name field too, because code might not be present
        // everything except Firefox
        e.name === 'QuotaExceededError' ||
        // Firefox
        e.name === 'NS_ERROR_DOM_QUOTA_REACHED') &&
      // acknowledge QuotaExceededError only if there's something already stored
      storage &&
      storage.length !== 0
    )
  }
}
