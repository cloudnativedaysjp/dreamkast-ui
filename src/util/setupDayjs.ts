import dayjs from 'dayjs'
import utc from 'dayjs/plugin/utc'
import timezone from 'dayjs/plugin/timezone'

export const setupDayjs = () => {
  dayjs.extend(timezone)
  dayjs.extend(utc)
  dayjs.tz.setDefault('Asia/Tokyo')
}
