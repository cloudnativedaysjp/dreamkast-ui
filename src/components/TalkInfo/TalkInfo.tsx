import React, { useEffect, useState, useCallback } from 'react'
import * as Styled from './styled'
import { Event, Talk, TrackApi, Configuration } from '../../client-axios'

type Props = {
  event?: Event
  selectedTalk?: Talk
  selectedTrackName?: string
  selectedTrackId?: number
}

export const TalkInfo: React.FC<Props> = ({
  event,
  selectedTalk,
  selectedTrackName,
  selectedTrackId,
}) => {
  const [timer, setTimer] = useState<number>()
  const [viewerCount, setViewerCount] = useState<string>()

  const twitterURL = (trackName?: string) => {
    const base = `http://twitter.com/share?url=https://event.cloudnativedays.jp/${event?.abbr}&related=@cloudnativedays&hashtags=${event?.abbr}`
    if (!trackName) return base
    return base + '_' + trackName
  }

  const getViewerCount = useCallback(async () => {
    console.log('Start getViewerCount')
    const api = new TrackApi(
      new Configuration({ basePath: window.location.origin }),
    )
    if (selectedTrackId) {
      try {
        console.log('Trying request')
        const { data } = await api.apiV1TracksTrackIdViewerCountGet(
          selectedTrackId.toString(),
        )
        setViewerCount(data.viewer_count.toString())
      } catch {
        setViewerCount('-')
      }
    }
  }, [viewerCount, selectedTrackId])

  useEffect(() => {
    getViewerCount()
    clearInterval(timer)
    setTimer(
      window.setInterval(() => {
        getViewerCount()
      }, 60 * 1000),
    )
  }, [selectedTrackId])

  return (
    <Styled.OuterContainer>
      <Styled.Container>
        {selectedTalk?.onAir && (
          <Styled.Live>LIVE üë• {viewerCount}</Styled.Live>
        )}
        <Styled.Title>{selectedTalk?.title}</Styled.Title>
        <Styled.SpeakerContainer>
          <Styled.Speaker>
            {selectedTalk?.speakers
              .map((speaker) => {
                return speaker.name
              })
              .join(' / ')}
          </Styled.Speaker>
          <div style={{ paddingRight: '20px' }} />
          {selectedTalk?.documentUrl && (
            <Styled.DocsLink href={selectedTalk?.documentUrl} target="_blank">
              ÁôªÂ£áË≥áÊñô„ÅØ„Åì„Å°„Çâ
            </Styled.DocsLink>
          )}
        </Styled.SpeakerContainer>
        <Styled.Content>{selectedTalk?.abstract}</Styled.Content>
        <Styled.SocialHeader>
          <Styled.TalkIcon src={`/${event?.abbr}/ui/talk_icon.png`} />
          ‰∏ÄÁ∑í„Å´Áõõ„Çä‰∏ä„Åå„Çç„ÅÜ
        </Styled.SocialHeader>
        <Styled.ButtonContainer>
          <Styled.ButtonLink
            href={twitterURL(selectedTrackName)}
            target="_blank"
          >
            <Styled.TweetButton>
              <Styled.TwitterImg src={`/${event?.abbr}/ui/twitter_logo.png`} />
              {`tweet #${event?.abbr}_${selectedTrackName}`}
            </Styled.TweetButton>
          </Styled.ButtonLink>
        </Styled.ButtonContainer>
      </Styled.Container>
    </Styled.OuterContainer>
  )
}
