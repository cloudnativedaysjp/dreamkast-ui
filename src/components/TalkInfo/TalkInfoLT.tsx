import React from 'react'
import * as Styled from './styled'

type Props = {
  eventAbbr: string
  title: string
  content: string
}

export const TalkInfoLT: React.FC<Props> = ({ eventAbbr, title, content }) => {
  const xURL = `http://x.com/share?url=https://event.cloudnativedays.jp/${eventAbbr}&related=@cloudnativedays&hashtags=${eventAbbr}`
  return (
    <Styled.Container>
      <Styled.Title>{title}</Styled.Title>
      <Styled.Content>{content}</Styled.Content>
      <Styled.SocialHeader>
        <Styled.TalkIcon src={`/${eventAbbr}/ui/talk_icon.png`} />
        一緒に盛り上がろう
      </Styled.SocialHeader>
      <Styled.ButtonContainer>
        <Styled.ButtonLink href={xURL} target="_blank">
          <Styled.XPostButton>
            <Styled.XImg src={`/${eventAbbr}/ui/x_logo.png`} />
            {`ポスト #${eventAbbr}`}
          </Styled.XPostButton>
        </Styled.ButtonLink>
      </Styled.ButtonContainer>
    </Styled.Container>
  )
}
