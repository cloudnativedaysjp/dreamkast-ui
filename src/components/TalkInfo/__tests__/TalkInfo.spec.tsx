import React from 'react'
import renderer from 'react-test-renderer'
import { MockEvent, Talks } from '../../../util/mock'
import { TalkInfo } from '../TalkInfo'

test('TalkInfo', () => {
  const component = renderer.create(
    <TalkInfo event={MockEvent} selectedTalk={Talks[0]} />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
