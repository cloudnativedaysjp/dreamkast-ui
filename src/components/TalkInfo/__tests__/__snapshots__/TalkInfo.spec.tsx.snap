// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TalkInfo 1`] = `
<div
  className="sc-gsTCUz gCgpQH"
>
  <div
    className="sc-bdfBwQ MrQxr"
  >
    <h2
      className="sc-dlfnbm EfhGP"
    >
      ほげふが
    </h2>
    <div
      className="sc-eCssSg dAwEgz"
    >
      <div
        className="sc-jSgupP kPMSyW"
      >
        alice / bob
      </div>
      <div
        style={
          Object {
            "paddingRight": "20px",
          }
        }
      />
    </div>
    <div
      className="sc-gKsewC kXjrHy"
    >
      ぴよぴよ
    </div>
    <div
      className="sc-fubCfw iXuTbZ"
    >
      <img
        className="sc-pFZIQ ehvigS"
        src="/o11y2022/ui/talk_icon.png"
      />
      一緒に盛り上がろう
    </div>
    <div
      className="sc-jrAGrp iSvhwg"
    >
      <a
        className="sc-kEjbxe cgtSAu"
        href="http://twitter.com/share?url=https://event.cloudnativedays.jp/o11y2022&related=@cloudnativedays&hashtags=o11y2022"
        target="_blank"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text sc-dQppl WRCRi"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <img
              className="sc-bqyKva cKheEB"
              src="/o11y2022/ui/twitter_logo.png"
            />
            tweet #o11y2022_undefined
          </span>
        </button>
      </a>
    </div>
  </div>
</div>
`;
